### PR TITLE
clean up links & spec docs

### LINKED ISSUE #1609: Update/remove consensus/README.md
this file: https://github.com/tendermint/tendermint/blob/master/consensus/README.md
hasn't been touched in nearly 2 years - probably out of date.

@ebuchman can it be `rm`'d?

### PR DESCRIPTION
**Pull request by zramsay**
closes #1609, #1617 

* [x] Updated all relevant documentation in docs
* [N/A] Updated all code comments where relevant
* [N/A] Wrote tests
* [N/A] Updated CHANGELOG.md

### CODE REVIEW

File: docs/spec/reactors/mempool/README.md [hunk 1]
```diff
@@ -1,11 +0,0 @@
-# Mempool Specification
```
Comment (melekes) [inline docs/spec/reactors/mempool/README.md]: why remove? was it moved somewhere?

Comment (zramsay) [reply docs/spec/reactors/mempool/README.md]: not moved. The only thing in this file is a ToC telling the reader about the 4 files in this directory. For consistency we should either have a README like this in every directory or none at all. The former option doesn't make much sense IMO

File: docs/spec/README.md [hunk 2]
```diff
@@ -14,21 +14,21 @@ please submit them to our [bug bounty](https://tendermint.com/security)!
 
 ### Data Structures
 
-- [Encoding and Digests](./blockchain/encoding.md)
-- [Blockchain](./blockchain/blockchain.md)
-- [State](./blockchain/state.md)
+- [Encoding and Digests](https://github.com/tendermint/tendermint/blob/master/docs/spec/blockchain/encoding.md)
```
Comment (ebuchman) [inline docs/spec/README.md]: oh goodness! I'm sorry, by absolute I meant with respect to the repo. so just `/docs/spec/...` is fine. as long as the first `/` means `top of this repo`. Full links is probably excessive

File: docs/spec/reactors/mempool/README.md (hunk 1 above)
Comment (ebuchman) [reply docs/spec/reactors/mempool/README.md]: maybe we can at least try to pattern all the reactor dirs the same way, so its always like `config.md, messages.md, state.md` or something

**Pull request merged**